llm_bioinformatics_research/src/Testing/ForgotPassword.test.js

```diff
@@ -0,0 +1,124 @@
+import ForgotPassword from "../Components/ForgotPassword/ForgotPassword";
+
+global.fetch = jest.fn();
+
+describe("ForgotPassword Component", () => {
+  beforeEach(() => {
+    fetch.mockClear();
+  });
+
+  it("renders the Forgot Password form", () => {
+    render(
+      <Router>
+        <ForgotPassword />
+      </Router>
+    );
+    expect(screen.getByText("Forgot Password")).toBeInTheDocument();
+    expect(screen.getByPlaceholderText("Email")).toBeInTheDocument();
+    expect(screen.getByText("Send email")).toBeInTheDocument();
+  });
+
+  it("updates email state on input change", () => {
+    render(
+      <Router>
+        <ForgotPassword />
+      </Router>
+    );
+    const emailInput = screen.getByPlaceholderText("Email");
+    fireEvent.change(emailInput, { target: { value: "test@example.com" } });
+    expect(emailInput.value).toBe("test@example.com");
+  });
+
+  it("displays success message when email is sent successfully", async () => {
+    fetch.mockResolvedValueOnce({
+      status: 200,
+      json: async () => ({ message: "Password reset email sent" }),
+    });
+
+    render(
+      <Router>
+        <ForgotPassword />
+      </Router>
+    );
+    const emailInput = screen.getByPlaceholderText("Email");
+    const submitButton = screen.getByText("Send email");
+
+    fireEvent.change(emailInput, { target: { value: "test@example.com" } });
+    fireEvent.click(submitButton);
+
+    const successMessage = await screen.findByText("Password reset email sent");
+    expect(successMessage).toBeInTheDocument();
+  });
+
+  it("displays error message when email is not found", async () => {
+    fetch.mockResolvedValueOnce({
+      status: 404,
+      json: async () => ({ message: "Email not found" }),
+    });
+
+    render(
+      <Router>
+        <ForgotPassword />
+      </Router>
+    );
+    const emailInput = screen.getByPlaceholderText("Email");
+    const submitButton = screen.getByText("Send email");
+
+    fireEvent.change(emailInput, { target: { value: "unknown@example.com" } });
+    fireEvent.click(submitButton);
+
+    const errorMessage = await screen.findByText("Email not found");
+    expect(errorMessage).toBeInTheDocument();
+  });
+
+  it("displays error message when token is expired", async () => {
+    fetch.mockResolvedValueOnce({
+      status: 400,
+      json: async () => ({ message: "Token expired. Please request a new password reset link." }),
+    });
+  
+    render(
+      <Router>
+        <ForgotPassword />
+      </Router>
+    );
+    const emailInput = screen.getByPlaceholderText("Email");
+    const submitButton = screen.getByText("Send email");
+  
+    fireEvent.change(emailInput, { target: { value: "test@example.com" } });
+    fireEvent.click(submitButton);
+  
+    const expiredTokenMessage = await screen.findByText("Token expired. Please request a new password reset link.");
+    expect(expiredTokenMessage).toBeInTheDocument();
+  });
+
```

Comment:
For this test, how are we ensuring that the link has expired? Is it simply just the mocking the error 400? Can we somehow check the validity of the link by employing a timer to expiration? This is just a suggestion.